tags/TagMap.md

```diff
@@ -0,0 +1,122 @@
+according to unique value of the `Tag`s. The `Tag`s can also be used to filter (include/exclude)
+measurements in a `View`. `Tag`s can further be used for logging and tracing.
+
+# Tag
+A `Tag` consists of TagScope, TagKey, and TagValue.
+
+## TagKey
+
+`TagKey` is the name of the Tag. TagKey along with `TagValue` is used to aggregate
+and group stats, annotate traces and logs.
+
+**Restrictions**
+- Must contain only printable ASCII (codes between 32 and 126 inclusive)
+- Must have length greater than zero and less than 256.
+- Must not be empty.
+
+## TagValue
+
+`TagValue` is a string. It MUST contain only printable ASCII (codes between
+32 and 126)
+
+## TagScope
+
+`TagScope` is used to determine the scope of a `Tag`. The values for the `TagScope` are
+Local or Global. In future, additional values can be added to address specific situations.
+
+The tag creator determines the scope of the tag.
+
+**Local Scope**
+Tag with `Local` scope are used within the process it created. Such tags are not propagated
+across process boundaries. Even if the process is reentrant the tag MUST be excluded from
+propagation when the call leaves the process.
+
+**Global Scope**
+If a tag is created with the `Global` scope then it is propagated across process boundaries subject 
+to outgoing and incoming (on remote side) filter criteria. See `TagFilter` in [Tag Propagation](#Tag Propagation).
+Typically 'Global' tags represents a transaction or request, processing of which spans multiple entities.
+
+# TagMap 
+`TagMap` is an abstract data type that represents collection of tags. 
+i.e., each key is associated with exactly one value.  `TagMap` is serializable, and it represents
+all of the information that could be propagated inside the process and across process boundaries.  
+`TagMap` is a recommended name but languages can have more language specific name.
+
+## Limits
+Combined size of all `Tag`s should not exceed 8192 bytes before encoding.
+The size restriction applies to the deserialized tags so that the set of decoded
+ `TagMap`s is independent of the encoding format.
+
+## TagMap Propagation
+`TagMap` may be propagated across process boundaries or across any arbitrary boundaries for various 
+reasons. For example, one may propagate 'project-id' Tag across all micro-services to break down metrics
+by 'project-id'. Not all `Tag`s in a `TagMap` should be propagated and not all `Tag`s in a `TagMap`
+should be accepted from a remote peer. Hence, `TagMap` propagator must allow specifying an optional
+list of ordered `TagFilter`s for receiving `Tag`s or for forwarding `Tag`s or for both. A `TagFilter` list 
+for receiving MAY be different then that for forwarding.
+
+If no filter is specified for receiving then all `Tag`s are received. 
+If no filter is specified for forwarding then all `Tag`s are forwarded except those that have `Local Scope`.
+
+### TagFilter
```

Comment:
We've seen some use-cases where filters are good for Views as well. Build distribution X (with tags x,y,z) where tagKey t has value foo. So that may be a different type of TagFilter, but want to say about that use-case because that may influence the name of this field. Maybe consider TagPropagationFilter?